apollo-router/src/router_factory.rs

```diff
@@ -150,11 +153,80 @@ impl RouterSuperServiceFactory for YamlRouterFactory {
         previous_router: Option<&'a Self::RouterFactory>,
         extra_plugins: Option<Vec<(String, Box<dyn DynPlugin>)>>,
     ) -> Result<Self::RouterFactory, BoxError> {
+        // we have to create afirst telemetry plugin before creating everything else, to generate a trace
+        // of router and plugin creation
+        let plugin_registry = &*crate::plugin::PLUGINS;
+        let mut initial_telemetry_plugin = None;
+
+        if previous_router.is_none() && apollo_opentelemetry_initialized() {
+            if let Some(factory) = plugin_registry
+                .iter()
+                .find(|factory| factory.name == "apollo.telemetry")
+            {
+                let mut telemetry_config = configuration
+                    .apollo_plugins
+                    .plugins
+                    .get("telemetry")
+                    .cloned();
+                if let Some(plugin_config) = &mut telemetry_config {
+                    inject_schema_id(Some(&Schema::schema_id(&schema)), plugin_config);
+                    match factory
+                        .create_instance(
+                            plugin_config,
+                            Arc::new(schema.clone()),
+                            configuration.notify.clone(),
+                        )
+                        .await
+                    {
+                        Ok(mut plugin) => {
+                            if let Some(telemetry) = plugin
+                                .as_any_mut()
+                                .downcast_mut::<crate::plugins::telemetry::Telemetry>(
+                            ) {
+                                telemetry.activate();
+                            }
+                            initial_telemetry_plugin = Some(plugin);
+                        }
+                        Err(e) => return Err(e),
+                    }
+                }
+            }
+        }
+
+        // for now we need the root span to be named request, otherwise the trace is not registered
+        let router_span = tracing::info_span!(STARTING_SPAN_NAME);
+        Self.inner_create(
+            configuration,
+            schema,
+            previous_router,
+            initial_telemetry_plugin,
+            extra_plugins,
+        )
+        .instrument(router_span)
+        .await
+    }
+}
+
+impl YamlRouterFactory {
+    async fn inner_create<'a>(
+        &'a mut self,
+        configuration: Arc<Configuration>,
+        schema: String,
+        previous_router: Option<&'a RouterCreator>,
+        initial_telemetry_plugin: Option<Box<dyn DynPlugin>>,
+        extra_plugins: Option<Vec<(String, Box<dyn DynPlugin>)>>,
+    ) -> Result<RouterCreator, BoxError> {
+        let query_planner_span = tracing::info_span!("query planner creation");
```

Comment:
```suggestion
        let query_planner_span = tracing::info_span!("query_planner_creation");
```

I think all of our spans are snake cased